internal/charmstore/store_test.go

```diff
@@ -1723,20 +1723,46 @@ func (s *StoreSuite) TestFindBestEntityWithMultiSeriesCharms(c *gc.C) {
 }
 
 var updateEntityTests = []struct {
+	about     string
 	url       string
+	update    bson.D
+	check     func(*gc.C, *mongodoc.Entity)
 	expectErr string
 }{{
-	url: "~charmers/trusty/wordpress-10",
+	about:  "successful update",
+	url:    "~charmers/trusty/wordpress-10",
+	update: bson.D{{"$set", bson.D{{"extrainfo.test", []byte("PASS")}}}},
+	check: func(c *gc.C, e *mongodoc.Entity) {
+		c.Assert(string(e.ExtraInfo["test"]), gc.Equals, "PASS")
+	},
 }, {
+	about:     "not found",
 	url:       "~charmers/precise/wordpress-10",
+	update:    bson.D{{"$set", bson.D{{"extrainfo.test", []byte("PASS")}}}},
 	expectErr: `cannot update "cs:precise/wordpress-10": not found`,
+}, {
+	about:  "empty update",
+	url:    "~charmers/trusty/wordpress-10",
+	update: bson.D{},
+	check: func(c *gc.C, e *mongodoc.Entity) {
+		c.Assert(e.PromulgatedURL, gc.Not(gc.IsNil))
+		c.Assert(e.PromulgatedURL.String(), gc.Equals, "cs:trusty/wordpress-4")
+	},
+}, {
+	about:  "nil update",
+	url:    "~charmers/trusty/wordpress-10",
+	update: nil,
+	check: func(c *gc.C, e *mongodoc.Entity) {
+		c.Assert(e.PromulgatedURL, gc.Not(gc.IsNil))
+		c.Assert(e.PromulgatedURL.String(), gc.Equals, "cs:trusty/wordpress-4")
+	},
 }}
```

Comment:
Do we have a test where we pass a bson with data which are unknown to charmstore ?